src/sentry/sentry_apps/apps.py

```diff
@@ -185,30 +185,29 @@ def _update_events(self) -> None:
             self.sentry_app.events = expand_events(self.events)
 
     def _update_service_hooks(self) -> None:
-        hooks = hook_service.update_webhook_and_events(
-            organization_id=self.sentry_app.owner_id,
-            application_id=self.sentry_app.application_id,
+        if self.sentry_app.is_published:
+            # if it's a published integration, we need to do many updates so we have to do it in a task so we don't time out
+            # the client won't know it succeeds but there's not much we can do about that unfortunately
+            create_or_update_service_hooks_for_sentry_app(
```

Comment:
@markstory So, I want this to run as an asynchronous control silo task but when I use `delay`, it doesn't work even though I specify the `silo_mode` as `CONTROL`. How do I do that?

Reply:
Odd. The task looks like it is correctly defined, and the module containing the task is already part of celery's imports. Do you get an error when the task runs, or when it is scheduled?

Reply:
@markstory well, I got it working now so I guess I'm good.